AutorestSwiftTest/AutoRestResourceFlatteningTest.swift

```diff
@@ -44,21 +44,304 @@ class AutoRestResourceFlatteningTest: XCTestCase {
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_putWrappedArray() throws {
+        let expectation =
+            XCTestExpectation(
+                description: "Call autorestresourceflatteningtestservice.putWrappedArray succeed"
+            )
+        let wrapped = [
+            WrappedProduct(value: "test")
+        ]
+        client.autorestresourceflatteningtestservice.put(wrappedArray: wrapped) { result, httpResponse in
+            switch result {
+            case .success:
+                expectation.fulfill()
+            case let .failure(error):
+                let details = self.errorDetails(for: error, withResponse: httpResponse)
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.putWrappedArray failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_putSimpleProduct() throws {
+        let expectation =
+            XCTestExpectation(
+                description: "Call autorestresourceflatteningtestservice.putSimpleProduct succeed"
+            )
+        let product = SimpleProduct(
+            maxProductDisplayName: nil,
+            genericValue: nil,
+            odataValue: nil,
+            productId: "Product1",
+            description: "Description"
+        )
+        client.autorestresourceflatteningtestservice.put(simpleProduct: product) { result, httpResponse in
+            switch result {
+            case .success:
+                expectation.fulfill()
+            case let .failure(error):
+                let details = self.errorDetails(for: error, withResponse: httpResponse)
+                print("test failed. error=\(details)")
+                XCTFail("Call autorestresourceflatteningtestservice.putSimpleProduct failed")
+            }
+        }
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func test_resourceFlattening_putSimpleProductWithGroupingFlattenParameterGroup() throws {
+        let expectation =
+            XCTestExpectation(
+                description: "Call autorestresourceflatteningtestservice.putSimpleProductWithGroupingFlattenParameterGroup succeed"
+            )
+        let group = FlattenParameterGroup(
+            name: "name",
+            simpleBodyProduct: SimpleProduct(
+                maxProductDisplayName: nil,
+                genericValue: nil,
+                odataValue: nil,
+                productId: "Product1",
+                description: "Description"
+            ),
+            productId: "Product1",
+            description: nil,
+            maxProductDisplayName: nil,
+            genericValue: nil,
+            odataValue: "Description"
+        )
+        client.autorestresourceflatteningtestservice
+            .putSimpleProductWithGrouping(flattenParameterGroup: group) { result, httpResponse in
```

Comment:
This seems like a failure of the naming heuristics. I would think this signature should look like:
`client.blah.putSimpleProduct(withGrouping: group)`

@sacheu, what do you think?

Reply:
I think `client.blah.putSimpleProduct(withGrouping: group)` make more sense.

Reply:
I agree, but I think I will separate this into a different issue, or let it be part of the PR for `test_resourceFlattening_putSimpleProductWithGroupingFlattenParameterGroup`